app/Http/Livewire/Dashboard/Post/LivewireTable.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace App\Http\Livewire\Dashboard\Post;
+
+use Auth;
+use App\Models\Post;
+use Livewire\Component;
+use WireUi\Traits\Actions;
+use Livewire\WithPagination;
+use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
+
+class TableLivewire extends Component
+{
+    use WithPagination, Actions, AuthorizesRequests;
+
+    public $perPage = 10;
+    public $sortBy = 'updated_at';
+    public $orderBy = 'DESC';
+    public $search = '';
+    public $checked = [];
+    public $is_checked_all = false;
+    public $action = '';
+
+    protected $listeners = ['executeAction'];
+
+    // called eachtime an event happens
+    public function render()
+    {
+        return view('livewire.dashboard.post.table-livewire',  [
+            'posts' => $this->posts,
+        ]);
+    }
+
+    // https://laravel-livewire.com/docs/2.x/lifecycle-hooks#class-hooks
+    public function updating($name, $value)
+    {
+        switch ($name) {
+            case 'search':
+            case 'perPage':
+                $this->resetPage();
+                break;
+        }
+    }
+
+    public function sortBy($field)
+    {
+        if ($this->sortBy === $field) {
+            $this->orderBy = $this->orderBy === 'ASC' ? 'DESC' : 'ASC';
+        } else {
+            $this->orderBy = 'ASC';
+        }
+
+        $this->sortBy = $field;
+    }
+
+    public function checkAll()
+    {
+        if ($this->is_checked_all) {
+            $this->checked = $this->posts->pluck('id')->all();
+        } else {
+            $this->checked = [];
+        }
```

Comment:
@coderabbitai I prefer avoiding else blocks in such way:
```suggestion
        $this->checked = [];
        
        if ($this->is_checked_all) {
            $this->checked = $this->posts->pluck('id')->all();
        }
```